### ISSUE TITLE
Support terraform-provider-aws 4.x

### ISSUE THREAD

**Issue by mgerlach** [terraform-aws-modules/terraform-aws-s3-bucket#137]
## Is your request related to a new offering from AWS?
No

## Is your request related to a problem? Please describe.
https://github.com/hashicorp/terraform-provider-aws 4.0.0 has breaking changes for the S3 bucket resource, see https://github.com/hashicorp/terraform-provider-aws/releases/tag/v4.0.0

## Describe the solution you'd like.
This module should work with https://github.com/hashicorp/terraform-provider-aws 4.x.x

## Describe alternatives you've considered.
We could abandon use of this module, but really don't want to ;)

## Additional context
N/A

**Comment by dzzun-ops** [comment]
also faced with this problem:
`Could not retrieve the list of available versions for provider hashicorp/aws: no available releases match the given constraints ~> 3.69, 4.0.0`

**Comment by kleytonhsantos** [comment]
This is the real problem:

![Screenshot from 2022-02-10 20-35-38](https://user-images.githubusercontent.com/11460558/153590689-e58629a2-a417-4d6c-a9cd-0301bf07339a.png)

To solve the problem force the version of provider to the previous version, in my case I used 3.74.1 and this was the result:
![Screenshot from 2022-02-10 20-34-52](https://user-images.githubusercontent.com/11460558/153591462-d65c79a5-a45c-4f7f-bbf7-9f1d0d67c595.png)

**Comment by jdiebold** [comment]
They pulled stuff like logging and lifecycle rule out of the bucket resource into separate resources. That's why it's the values are unconfigurable.

**Comment by mgerlach** [comment]
People will eventually want to use version 4.x of the provider.

**Comment by antonbabenko** [comment]
Thanks for opening this issue. We are well aware of version 4.x and we will work on the support in the coming weeks.

**Comment by antonbabenko** [comment]
For the time being, please use version 2.14.1 of this module where Terraform AWS provider version is locked to [`~> 3.69`](https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/pull/136/files#diff-dfd5848fa77a04d0343891fe859286cc210473d10f0e62c7f99f0d5ecf1a9927R7)

**Comment by baronnenate** [comment]
how do you force the use of a previous version of a module?

**Comment by baronnenate** [comment]
I've set the provider version as follows:
```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.69"
    }
  }

  required_version = "~> 1.1.4"
}
```

...and I have set the S3 module as follows:
```
module "s3_playbooks" {
  source = "terraform-aws-modules/s3-bucket/aws"
  version = "2.14.1"
```

But am still getting the error:
Initializing provider plugins...
- Finding hashicorp/aws versions matching ">= 3.62.0, >= 3.63.0, ~> 3.69, >= 3.72.0, >= 4.0.0"...
- Finding hashicorp/random versions matching ">= 3.1.0"...
- Installing hashicorp/random v3.1.0...
- Installed hashicorp/random v3.1.0 (signed by HashiCorp)
╷
│ Error: Failed to query available provider packages
│
│ Could not retrieve the list of available versions for provider hashicorp/aws: no available releases match the given constraints >= 3.62.0, >= 3.63.0, ~> 3.69, >=       
│ 3.72.0, >= 4.0.0

**Comment by antonbabenko** [comment]
You have other modules in this configuration that require `>= 4.0.0`, so it makes it impossible to find a single version of the AWS provider.

**Comment by baronnenate** [comment]
ah ok thanks anton, so shall I force those other modules to use an older version?  (they're the EC2 modules)

**Comment by antonbabenko** [comment]
Yes, I think it would be easier than trying to use both version 3 and version 4 together in the same project as described [here](https://www.terraform.io/language/providers/configuration#alias-multiple-provider-configurations).

Version 3 could be installed under another `alias` and passed into this module, like this (I didn't try this):
```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.69"
      configuration_aliases = [ aws.ancient ]
    }
  }
}

provider "aws" {
  alias  = "ancient"
  region = "us-west-2"
}

module "s3_playbooks" {
  source = "terraform-aws-modules/s3-bucket/aws"
  version = "2.14.1"

  providers = {
    aws = aws.ancient
  }
}
```

**Comment by rjshrjndrn** [comment]
> ```terraform
> terraform {
>   required_providers {
>     aws = {
>       source  = "hashicorp/aws"
>       version = "~> 3.69"
>       configuration_aliases = [ aws.ancient ]
>     }
>   }
> }
> ```

even this is not working.. still gettig error
![image](https://user-images.githubusercontent.com/2563385/156429359-92fcdcea-a21e-4ddf-8a25-6d38b55b7ea3.png)

**Comment by goural** [comment]
I am using     aws = {  version = "~> 3.69"  }

Terraform_version = ">= 0.14"
--
Getting these warnings -
**│ Warning: Argument is deprecated**
│ 
│   with module.s3Bucket.aws_s3_bucket.this,
│   on .terraform/modules/main.tf line 7, in resource "aws_s3_bucket" "this":
│    7: resource "aws_s3_bucket" "this" {
│ 
**│ Use the aws_s3_bucket_website_configuration resource instead
│ 
│ (and 13 more similar warnings elsewhere)**
╵

**Comment by pfilbin90** [comment]
Hello, can we get an update please on the status of this?  I would very much like to upgrade to provider version 4

**Comment by antonbabenko** [comment]
This issue has been resolved in version 3.0.0 :tada:
